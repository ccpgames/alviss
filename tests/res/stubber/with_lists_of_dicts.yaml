foo:
  bar:
    list_of_people:  # This is all optional
      - name: str
        age: int
        shoe_size: float
        favorite_foods:
          - str
        health:
          blood_pressure: int
          weight: float
          alive: bool
    list_of_important_people:
      - name: str*  # This make all of its ancestors Required!
        title: str