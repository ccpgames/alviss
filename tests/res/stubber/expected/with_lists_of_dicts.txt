__all__ = [
    'CfgFooBarListOfPeopleHealthStub',
    'CfgFooBarListOfPeopleStub',
    'CfgFooBarListOfImportantPeopleStub',
    'CfgFooBarStub',
    'CfgFooStub',
    'CfgStub',
    'AlvissConfigStub',
]

from typing import *
from alviss.structs import Empty
from alviss.structs.cfgstub import _BaseCfgStub
from alviss.structs import BaseConfig


class CfgFooBarListOfPeopleHealthStub(_BaseCfgStub):
    blood_pressure: Union[int, Empty]
    weight: Union[float, Empty]
    alive: Union[bool, Empty]


class CfgFooBarListOfPeopleStub(_BaseCfgStub):
    name: Union[str, Empty]
    age: Union[int, Empty]
    shoe_size: Union[float, Empty]
    favorite_foods: Union[List[str], Empty]
    health: Union[CfgFooBarListOfPeopleHealthStub, Empty]


class CfgFooBarListOfImportantPeopleStub(_BaseCfgStub):
    name: str
    title: Union[str, Empty]


class CfgFooBarStub(_BaseCfgStub):
    list_of_people: Union[List[CfgFooBarListOfPeopleStub], Empty]
    list_of_important_people: List[CfgFooBarListOfImportantPeopleStub]


class CfgFooStub(_BaseCfgStub):
    bar: CfgFooBarStub


class CfgStub(_BaseCfgStub):
    foo: CfgFooStub


class AlvissConfigStub(BaseConfig, CfgStub):
    pass
